pages/index.tsx

```diff
@@ -42,26 +18,30 @@ const Home = ({
         <Banner />
         <Divider marginTop="15px" />
         <SearchInput />
-        {products.map((product) => {
-          return (
-            <Fragment key={product.id}>
-              <ProductCard productInfo={product} />
-              <Divider />
-            </Fragment>
-          );
-        })}
-        <Button
-          alignSelf="center"
-          w="100px"
-          marginTop="20px"
-          borderRadius="30px"
-          color="white"
-          backgroundColor="brand.primary-900"
-          isLoading={isMoreButtonLoading}
-          onClick={() => handleMoreProductClick()}
-        >
-          <DownloadIcon w="5" h="5" />
-        </Button>
+        {productPages &&
+          productPages.pages.map((page) => {
+            return page.data.map((product) => {
+              return (
+                <Fragment key={product.id}>
+                  <ProductCard productInfo={product} />
+                  <Divider />
+                </Fragment>
+              );
+            });
+          })}
```

Comment:
아래 코드를 자주 사용하기 때문에 컴포넌트로 분리한다면
```js
 <Fragment key={product.id}>
  <ProductCard productInfo={product} />
  <Divider />
</Fragment>
```

이렇게만 되어도 깔끔해 보이지 않을까 싶습니다! 중간에 page에서도 구조 분해로 data도 빼보았습니다!
```suggestion
        {productPages &&
          productPages.pages.map(({ data }) => {
            return data.map((product) => <ProductCardContainer>);
          })}
```

Reply:
요렇게도 될까요?

```js
    {productPages?.pages.map(({ data }) =>
      data.map((product) => <ProductCardContainer {product} />)
    )}
```

Reply:
그저 갓호석이네요

Reply:
오 옵셔널체이닝을 적극적으로 사용하겠습니다!!

또한 분리된 `ProductCardContainer`에 내용이 짧지만 여러 목록을 보여주는 페이지에서 공통적으로 쓰일테니 전체적인 중복코드를 줄여줄 것 같습니다!!
모두 소중한 의견을 내주셔서 생각이 정리될 수 있었습니다! 감사합니다!!